### PR TITLE
test: refactor ls to use AVA

### PR DESCRIPTION
**Pull request by nfischer**
Will update when I verify all tests are copied

### CODE REVIEW

**Comment by nfischer** [comment]
Ready for review

File: ava-test/ls.js [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+  let result = shell.ls('-l', 'resources/ls');
+  t.falsy(shell.error());
+  t.is(result.code, 0);
+  let idx = -1;
+  for (let k = 0; k < result.length; k++) {
+    if (result[k].name === 'file1') {
+      idx = k;
+      break;
+    }
+  }
```
Comment (freitagbr) [inline ava-test/ls.js]: How about this:
```js
const idx = result.map(r => r.name).indexOf('file1');
```

File: ava-test/ls.js [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+  let result = shell.ls('-lR', 'resources/ls/');
+  t.falsy(shell.error());
+  t.is(result.code, 0);
+  let idx = -1;
+  for (let k = 0; k < result.length; k++) {
+    if (result[k].name === 'a_dir/b_dir') {
+      idx = k;
+      break;
+    }
+  }
```
Comment (freitagbr) [inline ava-test/ls.js]: Same here:
```js
const idx = result.map(r => r.name).indexOf('a_dir/b_dir');
```

File: ava-test/ls.js (hunk 1 above)
Comment (nfischer) [inline ava-test/ls.js]: Done

File: ava-test/ls.js (hunk 2 above)
Comment (nfischer) [inline ava-test/ls.js]: Done

**Comment by nfischer** [comment]
PTAL

**Pull request merged**